galahq/gala
# Heroku maintenance, upgrade to Postgres 16 and remove redis and sidekiq dependencies

The Postgres and Redis instances require maintenance.

> Heroku Postgres 13 is being deprecated on October 29, 2024 - [heroku.com/changelog-items/2778](https://devcenter.heroku.com/changelog-items/2778).
> Heroku Redis 6 is being deprecated on Dec 2, 2024 and requires TLS - [heroku.com/changelog-items/2992](https://devcenter.heroku.com/changelog-items/2992)

This PR upgrades Postgres and removes dependency on Redis and Sidekiq.

Deployed changes:

- followed Heroku recommendation to create a new migration that alters the case_search_index materialized view to be compatible with Postgres 16
- updated docker-compose.yml file from Postgres 13->16
- changed background jobs to run `:inline` (synchronously) and remove dependency on Redis and Sidekiq
- added the rake test:unit to run only the unit tests


To test:

1. prune docker, (docker compose down, docker system prune, docker volume prune, docker volume rm gala_db_data)
1. start app with docker compose up --build
1. ensure app builds and can be accessed at http://localhost:3000, you should see seed data
1. run docker compose run web rake indices:refresh inside container to ensure it works
1. trying searching for a case and ensure you are able to see results
inspect docker console, ensure you see no Postgres or Redis errors while performing exploratory testing in the app
1. ensure unit tests still pass, `docker compose run web rake test:unit`
1. ensure copying a case works and creating a translated case works.
